Python/examples/isda-engine.py

```diff
@@ -72,10 +72,17 @@
                       ql.Thirty360(),isda_ibor)
     for i in range(len(swap_tenors))]
 
-discountCurve = ql.RelinkableYieldTermStructureHandle()
-discountCurve.linkTo(ql.PiecewiseLogLinearDiscount(0,ql.WeekendsOnly(),
-                                                   isdaRateHelpers,
-                                                   ql.Actual365Fixed()))
+# Get around following error
+# yield term structure reference date (May 25th, 2009 should be evaluation date (May 21st, 2009)
+spot_date = ql.WeekendsOnly().advance(tradeDate, 2 * ql.Period(ql.Daily))
+pw_flat_fwd = ql.PiecewiseFlatForward(spot_date, isdaRateHelpers, ql.Actual365Fixed())
```

Comment:
Thanks, Francis.  I seem to get the same results by passing `tradeDate` here as the reference date and passing `pw_flat_fwd` to `discountCurve` directly, instead of building the curve below.  Do you see any problems with this?

Reply:
That is cleaner. I was using a log linear discount curve in the first commit (to get a discount factor of 1 on spot date) and left in the redundant code where I extract the node values and construct a new curve.

For background, I was trying to being as close as possible to the ISDA CDS model document [here](https://www.cdsmodel.com/cdsmodel/assets/cds-model/docs/IBOR%20Interest%20Rate%20Curve%20Specification%20-%20All%20Currencies%20(March%2013,%202021).pdf). In particular, the discount factor on the spot date should be 1. The relevant excerpt is:

> Note that the curve construction as described sets the value of the discount factor for the spot date to 1. Where a value is required for a day prior to spot (e.g., the curve date itself), this is obtained by extrapolating at the forward rate between spot and the first deposit maturity – thus the discount factor for the curve date is actually slightly greater than 1. To the extent that a user requires a present value calculated to the date to which the curve applies, rather than the spot date, this can be obtained simply by dividing the spot value by the curve date discount factor.

I am not sure if this exact behaviour can be obtained currently using the existing QuantLib yield term structures that are allowed by the ISDA CDS engine.

Reply:
Hmm, using `ForwardCurve` will set discount=1 at trade and exp(-r*t) at spot.  If we want the discount to be 1 at spot I think the closest we can get is use a `DiscountCurve` and pass discount=1 for both trade and spot.  Unfortunately, as of now, `DiscountCurve` won't let us pass a discount != 1 at the reference date.

I'll try that and see what happens.

Reply:
Nope, it didn't work.  Results got worse.
I'll use the curve with reference = trade date.

Reply:
Yes, sorry, that was my first commit [here](https://github.com/lballabio/QuantLib-SWIG/pull/384/commits/b9156431d6a638f7d673d0717fb62d5ac5b7ea44) and then I switched to `ForwardCurve`. I think that I would need to do a bit more digging (ISDA CDS model code vs. QuantLib code) to get rid of the remaining small upfront differences.